core/src/main/java/org/fao/geonet/api/records/attachments/StoreUtils.java

```diff
@@ -130,23 +127,9 @@ public static void replaceDataDir(ServiceContext context, String sourceUuid, Str
 
             // In order to sync the 2 folders, we need to identify the records to be added, deleted and updated.
             List<MetadataResource> targetDeleteResources  = exceptMetadataResource(targetResources, sourceResources);
-            List<MetadataResource> targetAddResources = exceptMetadataResource(sourceResources, targetResources);
-            List<MetadataResource> targetUpdateResources = exceptMetadataResource(targetResources, targetDeleteResources);
-
-            // Add new records
-            for (MetadataResource resource: targetAddResources) {
-                try (Store.ResourceHolder holder = store.getResource(context, sourceUuid, visibility, resource.getFilename(), sourceApproved)) {
-                    store.putResource(context, targetUuid, holder.getPath(), visibility, targetApproved);
-                }
-            }
-
-            // update new records
 
-            for (MetadataResource resource: targetUpdateResources) {
-                try (Store.ResourceHolder holder = store.getResource(context, sourceUuid, visibility, resource.getFilename(), sourceApproved)) {
-                    store.putResource(context, targetUuid, holder.getPath(), visibility, targetApproved);
-                }
-            }
+            // copy records from source to target
+            store.copyResources(context, sourceUuid, targetUuid, visibility, sourceApproved, targetApproved);
```

Comment:
I'm not sure about this change. Seems the `Add new records` /  `update new records` blocks have been replaced by previous line, but I don't see the original logic is preserved.

Reply:
@josegar74 
I had concerns as well however after reviewing it, I believe it is OK.

The add and the update were basically doing the same thing by calling 
`store.putResource(context, targetUuid, holder.getPath(), visibility, targetApproved);`
for all the files that existed in the source folder.
The main difference in the previous logic is that it was separating them into insert and updates so that we could potentially call different methods in the future. But currently,  different logic currently does not exists for inserts and updates, so it looks like @wangf1122 simply did them all for one function.

The deletes are currently the only one that need to be done separately.

So the main thing that we are loosing from this change is the separation of the insert/updates - but since they were calling the same function, it was not really used.  We can always change it back in the future, if there is a need for using different functions for the inserts and updates.

Reply:
Indeed, thanks for the clarification.